source/rust_verify/src/rust_to_vir_base.rs

```diff
@@ -224,14 +224,21 @@ pub(crate) fn get_impl_paths<'tcx>(
         let inst_pred = &predicate_worklist[idx];
         idx += 1;
 
-        if let PredicateKind::Clause(Clause::Trait(_)) = inst_pred.kind().skip_binder() {
+        if let ClauseKind::Trait(_) = inst_pred.kind().skip_binder() {
             let poly_trait_refs = inst_pred.kind().map_bound(|p| {
-                if let PredicateKind::Clause(Clause::Trait(tp)) = &p {
-                    tp.trait_ref
-                } else {
-                    unreachable!()
-                }
+                if let ClauseKind::Trait(tp) = &p { tp.trait_ref } else { unreachable!() }
             });
+
+            // REVIEW: is erasing all regions okay?
+            let delegate = rustc_middle::ty::fold::FnMutDelegate {
+                regions: &mut |_| tcx.lifetimes.re_erased,
+                types: &mut |b| panic!("unexpected bound ty in binder: {:?}", b),
+                consts: &mut |b, ty| panic!("unexpected bound ct in binder: {:?} {}", b, ty),
+            };
+            let poly_trait_refs =
+                tcx.replace_escaping_bound_vars_uncached(poly_trait_refs, delegate);
+            let poly_trait_refs = poly_trait_refs.skip_binder();
```

Comment:
There are, apparently, non-late-bound regions in things like:
```
    pub fn as_ref(&self) -> (a: Option<&A>)
```
(the `Option<&a>`)

`codegen_select_candidate` now accepts a `TraitRef`, without a binder, and panics if it has escaping bound vars.